### ISSUE TITLE
Error in DeepARLightningModule.load_from_checkpoint

### ISSUE THREAD

**Issue by Serendipity31** [awslabs/gluon-ts#3014]
## Description
I am trying to run the code in the 'simple example' shown here: https://ts.gluon.ai/stable/index.html. 

The DeepAREstimator errors when it tries to complete the final epoch. Googling this error wasn't very fruitful. I was only able to find a discussion in the github for the Lightning-AI package regarding the phrasing of a similar error from load_from_checkpoint: https://github.com/Lightning-AI/lightning/issues/18169

When I rolled back to earlier versions of python (3.11.6 --> 3.11.5), gluonts (0.13.5 --> 0.13.4), torch (2.1.0 --> 2.0.1), and pytorch-lightning (2.1.0 --> 2.0.8), this error message disappeared (though the resulting forecasts as not at all what the example suggests they should be, and it's unclear if this is related or not).

## To Reproduce
The exact part of the example script that errors is: 
 
```python
model = DeepAREstimator(
    prediction_length=12, freq="M", trainer_kwargs={"max_epochs": 5}
).train(training_data)
```

## Error message or code output

```python
Traceback (most recent call last):
  File "/Users/usr/test.py", line 24, in <module>
    ).train(training_data)
      ^^^^^^^^^^^^^^^^^^^^
  File "/Users/usr/.virtualenvs/deepAR_python/lib/python3.11/site-packages/gluonts/torch/model/estimator.py", line 237, in train
    return self.train_model(
           ^^^^^^^^^^^^^^^^^
  File "/Users/usr/.virtualenvs/deepAR_python/lib/python3.11/site-packages/gluonts/torch/model/estimator.py", line 213, in train_model
    best_model = training_network.load_from_checkpoint(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/usr/.virtualenvs/deepAR_python/lib/python3.11/site-packages/pytorch_lightning/utilities/model_helpers.py", line 93, in __get__
    raise TypeError(
TypeError: The classmethod `DeepARLightningModule.load_from_checkpoint` cannot be called on an instance. Please call it on the class type and make sure the return value is used.
```

## Environment
- Operating system: Apple M1 Sonoma 14.0
- Python version:      3.11.6
- GluonTS version:    0.13.5
- Torch:                      2.1.0
- Pytorch-Lightning: 2.1.0

**Comment by Serendipity31** [comment]
As a small update, I tried the same example on a PC. I tried to recreate the environment from my mac that led to the error disappearing (see below). On my PC, using these versions did *not* alleviate the error. 

- Operating system: Windows 10 Enterprise 22H2
- Python version: 3.11.5
- GluonTS version: 0.13.4
- Torch version: 2.0.1
- Pytorch-lightning version: 2.0.8

**Comment by lostella** [comment]
Hi @Serendipity31, thanks for reporting this. This is being fixed in #3013
